lib/redirect-agent.js

```diff
@@ -0,0 +1,173 @@
+const { InvalidArgumentError } = require('./core/errors')
+const util = require('./core/util')
+const EventEmitter = require('events')
+const { getGlobalAgent } = require('./agent')
+const assert = require('assert')
+
+const kMaxRedirections = Symbol('maxRedirections')
+const kAgent = require('agent')
+
+class RedirectAgent extends EventEmitter {
+  constructor ({ maxRedirections = 10, agent = getGlobalAgent() } = {}) {
```

Comment:
Just a confirmation question here: by doing this, you allow `maxRedirections` to be specified both at agent and single request level, isn't it?

Reply:
Correct.

Reply:
Much better than mine then! ^^